### PR TITLE
Gallery Slider Block SQONE-493

### PR DESCRIPTION
**Pull request by jamiepaul**
## What does this do/fix?

Adds Gallery Slider block as per [Figma designs](https://www.figma.com/file/Wj2g7HVyBQWYIuxgNKMCxf/SQ1-Theme?node-id=2551%3A0).

Gallery images can be shown with or without caption based on block setting. Image ratio block setting controls display and some swiper config:
* Fixed image ratio displays 16:9 images centered w/ a full bleed, infinite loop slider
* Variable image ratio displays each image at its own aspect ratio

Notes:
1. Looking for feedback/ideas on a simple display for variable image ratio gallery in mobile. Right now there's a display bug that squeezes the images in width so ones that might be wider than the viewport fit.
2. Because swiper JS isn't initialized in the block editor, some base styling was added to visualize the slider layout: [fixed image ratio screenshot](http://p.tri.be/n8ioh8), [variable image ratio screenshot](http://p.tri.be/lEaPEV)

## QA

Links to relevant docs
- [Link to Issue SQONE-493](https://moderntribe.atlassian.net/browse/SQONE-493)
- [Screencast of block implementation admin and FE](http://p.tri.be/ITOTLN)

### CODE REVIEW

File: wp-content/themes/core/components/blocks/gallery_slider/Gallery_Slider_Controller.php [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+
+		return $slider;
+	}
+
+	/**
+	 * Get Slides
+	 *
+	 * @return array
+	 */
+	public function get_slides(): array {
```
Comment (jamiepaul) [inline wp-content/themes/core/components/blocks/gallery_slider/Gallery_Slider_Controller.php:229]: @mmcachran can I get your eyes on this? I'm sure there's a more eloquent way of returning the slide markup. I used the Gallery Grid block as a base when I started this one and it returns an array_map. Sort of seemed like overkill here but also seems like a BE decision.

File: wp-content/plugins/core/src/Blocks/Types/Gallery_Slider/Gallery_Slider.php [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+	public const CAPTION_DISPLAY_HIDE = 'caption_display_hide';
+
+	public const SLIDESHOW = 'slideshow';
+
+	public function add_block() {
+		$this->set_block( new Block( self::NAME, [
+			'title'       => __( 'Gallery Slider', 'tribe' ),
+			'description' => __( 'A custom block by Modern Tribe', 'tribe' ), // TODO: describe the block
+			'icon'        => '<svg enable-background="new 0 0 146.3 106.3" version="1.1" viewBox="0 0 146.3 106.3" xml:space="preserve" xmlns="http://www.w3.org/2000/svg"><style type="text/css">.st0{fill:#16D690;}.st1{fill:#21A6CB;}.st2{fill:#008F8F;}</style><polygon class="st0" points="145.2 106.3 72.6 42.3 26.5 1.2 0 106.3"/><polygon class="st1" points="145.2 106.3 0 106.3 72.6 42.3 118.6 1.2"/><polygon class="st2" points="72.6 42.3 145.2 106.3 0 106.3"/></svg>', // TODO: set SVG icon
+			'keywords'    => [ __( 'gallery', 'slider', 'carousel', 'image', 'tribe' ) ], // TODO: select appropriate keywords
```
Comment (dpellenwood) [inline wp-content/plugins/core/src/Blocks/Types/Gallery_Slider/Gallery_Slider.php:37]: Recognizing that this is TODO only, but the current structure will break translations. Each key word should be init's own translation call, or they should all share the same translation string + the `'tribe' text domain as the second argument. 

Example: `__( 'gallery, slider, carousel, image', 'tribe' )`

File: wp-content/themes/core/components/blocks/gallery_slider/css/gallery_slider.pcss [hunk 3]
```diff
@@ -0,0 +1,110 @@
+/* -----------------------------------------------------------------------------
+ *
+ * Component: Gallery Slider
+ *
+ * ----------------------------------------------------------------------------- */
+
+.b-gallery-slider {
+	padding-bottom: var(--spacer-70); /* should blocks have vertical padding? */
```
Comment (dpellenwood) [inline wp-content/themes/core/components/blocks/gallery_slider/css/gallery_slider.pcss:8]: There are some spacing vars defined for this; but tbh, the whole system needs a spacing pass. you can leave this as-is.

**Review by dpellenwood — APPROVED** [review]
This looks awesome, Jamie. Thanks for tackling this!

**Comment by nickpelton** [comment]
Merged as part of PR829

**Pull request closed without merging**